diff --git a/go/private/sdk.bzl b/go/private/sdk.bzl
index dd78c776..2f92c953 100644
--- a/go/private/sdk.bzl
+++ b/go/private/sdk.bzl
@@ -208,6 +208,13 @@ def _detect_host_platform(ctx):
 
     elif ctx.os.name == "mac os x":
         host = "darwin_amd64"
+
+        res = ctx.execute(["uname", "-m"])
+        if res.return_code == 0:
+            uname = res.stdout.strip()
+            if uname == "arm64":
+                host = "darwin_arm64"
+
     elif ctx.os.name.startswith("windows"):
         host = "windows_amd64"
     elif ctx.os.name == "freebsd":
